### PR TITLE
Add & Process models

### PR DESCRIPTION
**Pull request by Shakir-Hussain12**
**For this PR, I :**
- Created models  `User`, `Post`, `Comment`, `Like`.
- Added associations for each Model.
- Ran different queries such as `find`, `create`, `update` etc.
- Added method to retrieve latest 3 posts for a given user.
- Added method to retrieve latest 5 comments for a given post.
- Added rack files to update count for each associated table.
- Added methods to increase count for every addition of `Comment`, `Post`, or `Like`.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: true
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+
+  after_create do
+    author.increment!(:posts_count)
+  end
```
Comment (alexpuente0) [inline app/models/post.rb]: - You can use the `after_save` Active Callback to call the method that updates the post counter, which should be a private method.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, foreign_key: 'post_id', counter_cache: true
+
+  after_create do
+    post.increment!(:comments_count)
+  end
```
Comment (alexpuente0) [inline app/models/comment.rb]: - You can use the `after_save` Active Callback to call the method that updates the post counter, which should be a private method.

**Review by alexpuente0 — APPROVED** [review]
Hi @Shakir-Hussain12, 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Well written and functional code

## Optional suggestions
- I left you a couple of suggestions to make your project more secure by using private methods.
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (Shakir-Hussain12) [reply app/models/post.rb]: oh okay, thanks for the suggestion. Will definitely do it for the next PR.

**Pull request merged**